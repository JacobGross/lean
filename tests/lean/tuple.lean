import data.prod
open nat prod

set_option pp.universes true

definition tuple (A : Type) (n : nat) : Type :=
nat.rec_on n A (λ n r, r × A)

check @tuple
