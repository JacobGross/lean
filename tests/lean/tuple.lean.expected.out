tuple.{l_1} : Type.{max 1 l_1} → ℕ → Type.{max 1 l_1}
